lib/ansible/plugins/callback/logdna.py

```diff
@@ -0,0 +1,147 @@
+        description: Whether to ignore errors on failing or not
+        env:
+          - name: ANSIBLE_IGNORE_ERRORS
+        default: False
+      conf_hostname:
+        required: False
+        description: Alternative Host Name
+        env:
+          - name: LOGDNA_HOSTNAME
+      conf_tags:
+        required: False
+        description: Tags
+        env:
+          - name: LOGDNA_TAGS
+'''
+
+import logging
+import json
+import pip
+try:
+    import logdna
+except ImportError:
+    pip.main(['install', 'logdna'])
+from logdna import LogDNAHandler
+from uuid import getnode
+from socket import gethostname
+
+from ansible.plugins.callback import CallbackBase
+
+# Getting MAC Address of system:
+
+
+def get_mac():
+    mac = "%012x" % getnode()
+    return ":".join(map(lambda index: mac[index:index + 2], xrange(int(len(mac) / 2))))
+
+# Getting hostname of system
+
+
+def get_hostname():
+    return str(gethostname()).split('.local')[0]
+
+# LogDNA Callback Module:
+
+
+class CallbackModule(CallbackBase):
+
+    CALLBACK_VERSION = 0.1
+    CALLBACK_TYPE = 'aggregate'
+    CALLBACK_NAME = 'logdna'
+    CALLBACK_NEEDS_WHITELIST = True
+
+    def __init__(self, display=None):
+
+        super(CallbackModule, self).__init__(display=display)
+
+        self.disabled = True
+        self.playbook_name = None
+        self.playbook = None
+        self.conf_key = None
+        self.plugin_ignore_errors = None
+        self.conf_hostname = None
+        self.conf_tags = None
+
+    def set_options(self, task_keys=None, var_options=None, direct=None):
+        super(CallbackModule, self).set_options(task_keys=task_keys, var_options=var_options, direct=direct)
+
+        self.conf_key = self.get_option('conf_key')
+        self.plugin_ignore_errors = self.get_option('plugin_ignore_errors')
+        self.conf_hostname = self.get_option('conf_hostname')
+        self.conf_tags = self.get_option('conf_tags')
+
+        if self.plugin_ignore_errors is None:
+            self.plugin_ignore_errors = False
+
+        if self.conf_hostname is None:
+            self.conf_hostname = get_hostname()
+
+        if self.conf_tags is None:
+            self.conf_tags = ['ansible']
+        else:
+            if type(self.conf_tags) is str:
+                self.conf_tags = self.conf_tags.split(',')
+            elif type(self.conf_tags) is not list:
+                self.conf_tags = [str(self.conf_tags)]
+
+        if self.conf_key is None:
+            self.disabled = True
+            self._display.warning('LogDNA Ingestion Key has not been provided!')
+        else:
+            self.disabled = False
+            self.log = logging.getLogger('logdna')
+            self.log.setLevel(logging.INFO)
+            self.options = {'hostname': self.conf_hostname, 'mac': get_mac(), 'index_meta': True}
+            self.log.addHandler(LogDNAHandler(self.conf_key, self.options))
+
+    def sendLog(self, host, category, logdata):
+        if not self.disabled:
```

Comment:
you should not need to checked disabled, as the plugin itself wont be called at all if true